src/renderer/components/+preferences/preferences-navigation/extensions-preferences-navigation-item.injectable.ts

```diff
@@ -3,49 +3,54 @@
+import { extensionRegistratorInjectionToken } from "../../../../extensions/extension-loader/extension-registrator-injection-token";
+import type { LensRendererExtension } from "../../../../extensions/lens-renderer-extension";
+import currentPathParametersInjectable from "../../../routes/current-path-parameters.injectable";
+import routeIsActiveInjectable from "../../../routes/route-is-active.injectable";
+import { preferenceNavigationItemInjectionToken } from "./preference-navigation-items.injectable";
 
-const extensionsPreferencesNavigationItemInjectable = getInjectable({
+const extensionPreferencesNavigationItemRegistratorInjectable = getInjectable({
   id: "extension-preferences-navigation-item",
 
   instantiate: (di) => {
-    const preferenceItems = di.inject(
-      extensionsPreferenceItemsInjectable,
-    );
-
-    const navigateToPreferenceTab = di.inject(
-      navigateToPreferenceTabInjectable,
-    );
-
-    const route = di.inject(
-      extensionPreferencesRouteInjectable,
-    );
-
-    const routeIsActive = di.inject(
-      routeIsActiveInjectable,
-      route,
-    );
-
-    return {
-      id: "extensions",
-      label: "Extensions",
-      navigate: navigateToPreferenceTab(route),
-      isActive: routeIsActive,
-
-      isVisible: computed(
-        () => preferenceItems.get().length > 0,
-      ),
-
-      orderNumber: 70,
+    return (ext) => {
+      const extension = ext as LensRendererExtension;
+      const navigateToExtensionPreferences = di.inject(
+        navigateToExtensionPreferencesInjectable,
+      );
+      const extensionHasPreferences = extension.appPreferences.length > 0;
+      const extensionHasGeneralPreferences = extension.appPreferences.some(preferences =>
+        !preferences.showInPreferencesTab,
+      );
+      const isVisible = computed(() => extensionHasPreferences && extensionHasGeneralPreferences);
+      const extensionRoute = di.inject(extensionPreferencesRouteInjectable);
+      const pathParameters = di.inject(currentPathParametersInjectable);
+      const routeIsActive = di.inject(routeIsActiveInjectable, extensionRoute);
+
+      const isActive = computed(() => routeIsActive.get() && pathParameters.get().extensionId === extension.sanitizedExtensionId);
+
+      const extensionInjectable = getInjectable({
+        id: `extension-preferences-navigation-item-${extension.sanitizedExtensionId}`,
+        instantiate: () => ({
+          id: `extension-${extension.sanitizedExtensionId}`,
+          label: `${extension.name}`,
+          navigate: () => navigateToExtensionPreferences(extension.sanitizedExtensionId),
+          isActive,
+          isVisible,
+          orderNumber: 20,
+          fromExtension: true,
+        }),
+        injectionToken: preferenceNavigationItemInjectionToken,
+      });
+
+      di.register(extensionInjectable);
```

Comment:
Haven't seen before. If we not register the injectable it won't work or what's for?

Reply:
I can only suggest but yes it doesn't work if we just return `getInjectable()` without `di.register`. @jansav Can you bring more clarification there?

Reply:
Sure. Calling `getInjectable` doesn't have any side-effects. It just gets the injectable as it says. Here we are late-registering more implementations for `preferenceNavigationItemInjectionToken`. 

This code is registrator which adapts current extension API to the implementation we are using in Open Lens. This is to allow us do the required refactorings OpenLens without breaking the extension API. In the future we will be removing all of these registrators and export direct interfaces.

Naming of this file should mention "registrator".

Reply:
Renamed the file.